### PR TITLE
Fix money object deprecation warnings

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

Fixes all `Money` deprecation warnings but:

```
[DEPRECATION] `Money.infinite_precision` is deprecated - use `Money.default_infinite_precision` instead
```

I couldn't quite figure out this one because we don't call `infinite_precision` in our codebase, so either the gem itself uses it or the problem is in another gem we use.

I fixed all the other deprecation warnings by adapting the option keys to the names `Money` expects, without touching any caller or app-level config setting resulting in a pure refactoring. This should then not introduce any regression.

Note, however, that now, outputting a money amount in HTML results in a bunch of `<span>` tags but that shouldn't change how we display them at all. 

I expect this to speed up the test suite considerably given these many warnings get printed every time a Money object is instantiated and this happens a lot in the build.

Highly related to https://github.com/openfoodfoundation/openfoodnetwork/issues/6752.

#### What should we test?

~~Ensure that `PUT "CURRENCY SYMBOL BEFORE OR AFTER DOLLAR AMOUNT?"` section on /admin/general_settings/edit keeps working.~~

See a961792a089058fe1054a0fe153aff8207e9989b. That covers us so a green build would be enough, although is worth it that you check the test @filipefurtad0 and confirm whether it's enough or not. IMO it is.

Besides, given this is so critical we could test that a EUR instance also works. I personally checked a shopfront before and after changing the currency symbol position.

#### Release notes

Fix most money gem deprecation warnings.
Changelog Category: Technical changes

### CODE REVIEW

File: app/models/spree/app_configuration.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ class AppConfiguration < Preferences::Configuration
     preference :checkout_zone, :string, default: nil
     preference :currency, :string, default: "USD"
     preference :currency_decimal_mark, :string, default: "."
-    preference :currency_symbol_position, :string, default: "before"
+    preference :currency_symbol_position, :string, default: "%u%n"
```
Comment (sauloperez) [inline app/models/spree/app_configuration.rb:45]: I'll also convert from `before` to `%u%n` from `Spree::Money` instead. This way there's no chance of breaking any instance's configuration or having to migrate this setting to the new format.

File: config/initializers/money.rb [hunk 2]
```diff
@@ -1 +1,2 @@
 Money.rounding_mode = BigDecimal::ROUND_HALF_EVEN
+Money.default_currency = Money::Currency.new('USD')
```
Comment (Matt-Yorkley) [inline config/initializers/money.rb:2]: Shouldn't this be default currency of the instance?

File: spec/lib/spree/money_spec.rb [hunk 3]
```diff
@@ -128,10 +118,14 @@
       expect(money.to_s).to eq("1.000.00 €")
     end
 
+    # rubocop:disable Layout/LineLength
     it "formats as HTML if asked (nicely) to" do
       money = Spree::Money.new(10)
       # The HTMLified version of the euro sign
-      expect(money.to_html).to eq("10.00&nbsp;&#x20AC;")
+      expect(money.to_html).to eq(
+        "<span&nbsp;class=\"money-whole\">10</span><span class=\"money-decimal-mark\">.</span><span class=\"money-decimal\">00</span> <span class=\"money-currency-symbol\">&#x20AC;</span>"
```
Comment (Matt-Yorkley) [inline spec/lib/spree/money_spec.rb:126]: I guess we should manually test places where money formatting is used, to make sure there's no regressions?

File: config/initializers/money.rb (hunk 2 above)
Comment (sauloperez) [reply config/initializers/money.rb:2]: Oops, yes. Good catch 😅

File: spec/lib/spree/money_spec.rb (hunk 3 above)
Comment (sauloperez) [reply spec/lib/spree/money_spec.rb:126]: I'll manually check but new `spans` with classes we don't use for styling should have no effect

File: lib/spree/money.rb [hunk 4]
```diff
@@ -47,5 +44,30 @@ def format(options = {})
...
+      case position.to_sym
+      when :before
+        '%u%n'
+      when :after
+        '%n %u'
+      else
+        raise 'Invalid symbol position'
+      end
+    end
   end
```
Comment (andrewpbrett) [inline lib/spree/money.rb:72]: Nice, I was wondering how to do this without migrating the actual preferences. 💯

**Review by andrewpbrett — APPROVED** [review]
This looks great once the USD default that Matt spotted is fixed. It will be really nice to not see these messages!

**Review by Matt-Yorkley — APPROVED** [review]
:ok_hand:

File: spec/lib/spree/money_spec.rb (hunk 3 above)
Comment (sauloperez) [reply spec/lib/spree/money_spec.rb:126]: It works but it's weird that `money` returns this broken tag :thinking: 

Screenshot from 2021-03-09 18-50-36

Comment (Matt-Yorkley) [reply spec/lib/spree/money_spec.rb:126]: Ugh... I don't like it.

Can we strip the tags on that output, so the previous specs for the output pass?

Comment (sauloperez) [reply spec/lib/spree/money_spec.rb:126]: fair enough.

File: lib/spree/money.rb [hunk 5]
```diff
@@ -28,13 +28,7 @@ def to_s
     end
 
     def to_html(options = { html_wrap: true })
-      output = @money.format(@options.merge(options))
-      if options[:html_wrap]
-        # 1) prevent blank, breaking spaces
-        # 2) prevent escaping of HTML character entities
-        output = output.sub(" ", "&nbsp;").html_safe
-      end
```
Comment (Matt-Yorkley) [inline lib/spree/money.rb:39]: Great! :fire:

I don't see anywhere that the `options` argument ever gets passed to this method, so we could even remove the argument entirely, but it's not important :ok_hand:

**Review by jibees — APPROVED** [review]
👌

**Comment by filipefurtad0** [comment]
Hey @sauloperez ,

Agree - the spec is enough to cover whether changing the currency placement settings work. Checked this manually as a double check, and whether it takes effect as well as the general display of the currency, on staging-UK (£):

- admin -> orders and related pages
- admin -> reports
- admin -> products

- user -> checkout
- user -> cart
- user -> shopfront
- user -> transactions
- user -> orders 

Looks good to go :+1:

**Pull request merged**